app/src/androidTest/java/com/github/polypoly/app/commons/MockDB.kt

```diff
@@ -14,6 +14,8 @@ class MockDB: IRemoteStorage {
     private val keysHierarchy: MutableMap<String, MutableList<String>> = mutableMapOf()
     private val data: MutableMap<String, Any> = mutableMapOf()
 
+    private val listeners: MutableMap<String, MutableList<Pair<String, (Any) -> Unit>>> = mutableMapOf()
```

Comment:
@Jharaxus 's work I'll let him respond

Reply:
I don't think so because, on the contrary to remoteDb where Firebase executes the callbacks by itself when necessary, here we have to do it by hand. Therefore, this structure makes it easier to iterate over the listeners to call when a value is set for a given key, compared to having a pair(key, tag).

Reply:
Oh okay I get it, in that point of view I agree :)